bb_proxy/src/conn.rs

```diff
@@ -534,6 +548,33 @@ impl<'a, S: PacketStream + Send + Sync> Conn<'a, S> {
   /// Generates the json status for the server
   fn build_status(&self) -> JsonStatus { (self.status_builder)(&self.icon, self.ver) }
 
+  /// Parse BungeeCord's player info from address string
+  fn read_bungeecord_info(&self, addr: String) -> Option<LoginInfo> {
+    let mut id = None;
+    let mut properties = None;
+
+    let mut i = 0;
+    for v in addr.split('\0') {
+      match i {
+        2 => { id = Some(UUID::from_str(v).ok()?) }
+        3 => { properties = Some(serde_json::from_str(v).ok()?) }
+        _ => {}
+      }
+
+      i += 1;
+    }
+
+    if id.is_none() {
+      return None
+    }
+
+    Some(LoginInfo {
+      id: id.unwrap(),
+      name: "".to_string(),
+      properties: properties.unwrap(),
+    })
```

Comment:
```suggestion
    id.map(move |id| LoginInfo {
      id,
      name: "".to_string(),
      properties: properties.unwrap(),
    })
```

use a map or a match here, and handle if `properties` is None.

Reply:
done